pkg/globalconfig/global_config.go

```diff
@@ -60,13 +60,20 @@ type GlobalConfig struct {
 	ProjectTldGlobal             string                  `yaml:"project_tld"`
 	XdebugIDELocation            string                  `yaml:"xdebug_ide_location"`
 	NoBindMounts                 bool                    `yaml:"no_bind_mounts"`
-	UseTraefik                   bool                    `yaml:"use_traefik"`
+	Router                       string                  `yaml:"router"`
```

Comment:
For a config expecting a group of predefined strings I'd use a typed string like I'm doing it in my change see https://github.com/ddev/ddev/pull/4904/files#diff-42da3ac5f97f8c2d00f51c6afe831dadaf9245cc93f258cee0e6f1a6ac2e7a06R38-R46

Reply:
Awesome! Never even heard of typed string before.  I see https://golangbyexample.com/string-constant-golang/#Typed_string_constant and other references; not good google hits though.

Reply:
AFAICT there's not a concept of "typed string", but rather a typed constant. But using it does not seem to constrain their use, it just gives the constant a type. I'm not at all sure that your example is constraining the uses. I do agree. that we could use a constant named Router, which is done for the *values* that can be assigned here, using https://github.com/ddev/ddev/pull/4983/files#diff-0caf521c3856fcd77e5a3e51d2c3cf9a925478fa8d1beb69aa654a8cff7fce7eR45-R48 and then the ValidRouterTypes function.

Reply:
I guess the compiler currently does not warn if a native string is used instead of the new type. This ofc would be great, well to have a warning in minimum. Also not sure about IDE support if they only show constants with the correct type and may also differ in various IDEs. So currently this is more like a cosmetic thing but it also helps to find usages of the new type. Maybe there will be more benefit in the future. Anyway it's a good practise.